src/highlight/HighlightAnnotations.tsx

```diff
@@ -98,7 +105,7 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
             {isCreating && <HighlightCreator className="ba-HighlightAnnotations-creator" />}
 
             {/* Layer 3a: Staged (unsaved) highlight target, if any */}
-            {isCreating && staged && (
+            {(isCreating || isPromoting) && staged && (
```

Comment:
It seems like for the purposes of `HighlightAnnotations`, `isCreating` is synonymous with `isPromoting` -- would it make sense to move that up to `HighlightContainer` such that `isCreating = getAnnotationMode(state) === Mode.HIGHLIGHT || getIsPromoting(state)`?

Reply:
What is the purpose of `isPromoting` at this point?

Reply:
Actually, `isCreating` is used alone to determine to show `HighlightCreator`. But it seems like we may remove `HighlightCreator` later. I think we can combine them together when we remove `HighlightCreator` in another PR.

Reply:
@jstoffan We don't change mode now, so isCreating is always false. We have to have `isPromoting` to control showing staged and popup

Reply:
Can we just do that now? It seems like `isPromoting` is duplicative state if we don't plan to have a "Creator" component for highlights.